transitions/core.py

```diff
@@ -686,9 +686,13 @@ def add_transition(self, trigger, source, dest, conditions=None,
             source = [s.name if self._has_state(s) else s for s in listify(source)]
 
         for s in source:
-            if self._has_state(dest):
-                dest = dest.name
-            t = self._create_transition(s, dest, conditions, unless, before,
+            if dest == '=':
+                real_dest = s
```

Comment:
This is because if you have more than one source (more than one iteration of the loop there), then after the first loop the value of dest is not = anymore and the destination for all created transitions is the first source state.